tools/sdk/esp32s2/include/arduino_tinyusb/tinyusb/src/class/hid/hid.h

```diff
@@ -300,6 +300,17 @@ typedef struct TU_ATTR_PACKED
   int8_t  pan;     // using AC Pan
 } hid_mouse_report_t;
 
+// Absolute Mouse data struct is a copy of the Standard (relative) Mouse Report
```

Comment:
any change to headers in the sdk folder should be done in the appropriate remote place. in his case, it is https://github.com/hathach/tinyusb

Reply:
let's get the descriptor merged into tinyusb and then we will discuss how to best plug it in. Probably separate class with own report ID

Reply:
These are separate classes: `USBHIDRelativeMouse` (aliased as `USBHIDMouse` for legacy), and `USBHIDAbsoluteMouse`.

There's also a base class `USBHIDMouseBase` but only to provide inheritance, so it's not an issue to have separate instances and should also work with a different report ID.

I tried to solve the build errors by using virtual / override, will do more testing.

Reply:
The [build failed](https://github.com/hathach/tinyusb/runs/5474344012?check_suite_focus=true) for ESP32 only, but looking at the error it's not coming from my pull request.

Will I have to do something to trigger the check_suite once this is fixed on arduino-esp32 repo?

Reply:
last build error points to a missing implementation for esp32s3, I can't remember why this PR was only targeting esp32s2, but I wonder how this should be dealt with: should I also apply the hid patch to sdk/tools/esp32s3/includes/arduino_tinyusb folder?

also the mouse class needs rewriting

Reply:
header patches should not be in the sdk folder at all. They should be in `USBHIDMouse.h`. In 3.0.0 SDK is no longer part of the repo.

Reply:
oh nice, that makes more sense, also it simplifies the path

I guess this PR is no longer relevant since it's based on the 2.x.x core, should I keep it open until I can properly test it with core 3.0.0?

Reply:
yes. It should work with 3.0.0 as well